### PR TITLE
ENH: Pandas dataframe chunksize

### PR DESCRIPTION
**Pull request by thequackdaddy**
Second try for #1034.

### CODE REVIEW

File: dask/dataframe/io.py [hunk 1]
```diff
@@ -338,8 +338,11 @@ def from_pandas(data, npartitions, sort=True):
     ----------
     df : pandas.DataFrame or pandas.Series
         The DataFrame/Series with which to construct a dask DataFrame/Series
-    npartitions : int
+    npartitions : int (defaults to None)
         The number of partitions of the index to create
+    chunksize : int (defaults to None)
+        The size of blocks to pull out from ctable.  Ideally as large as can
```
Comment (shoyer) [inline dask/dataframe/io.py]: ctable -> the dataframe

**Comment by thequackdaddy** [comment]
Thanks @shoyer

**Comment by thequackdaddy** [comment]
Just checking is there anything else y'all want done with this before it's merged? I think it needs to be rebased again, sadly. Considering I screwed it up last time, Ima let someone else do the honors. 

Thanks!

**Comment by mrocklin** [comment]
Sure.  I'll take care of this tomorrow.

File: dask/dataframe/io.py [hunk 2]
```diff
@@ -338,8 +338,11 @@ def from_pandas(data, npartitions, sort=True):
     ----------
     df : pandas.DataFrame or pandas.Series
         The DataFrame/Series with which to construct a dask DataFrame/Series
-    npartitions : int
+    npartitions : int (defaults to None)
```
Comment (sinhrks) [inline dask/dataframe/io.py]: Can you follow other style, ``int, default None`` (no parenthesis)

File: dask/dataframe/io.py [hunk 3]
```diff
@@ -381,8 +390,14 @@ def from_pandas(data, npartitions, sort=True):
     if not isinstance(data, (pd.Series, pd.DataFrame)):
         raise TypeError("Input must be a pandas DataFrame or Series")
 
+    if ((npartitions is None) + (chunksize is None)) in [0, 2]:
```
Comment (sinhrks) [inline dask/dataframe/io.py]: I prefer normal logical expression.

File: dask/dataframe/tests/test_io.py [hunk 4]
```diff
@@ -401,6 +401,11 @@ def test_from_pandas_dataframe():
     assert len(ddf.divisions) == len(ddf.dask) + 1
     assert type(ddf.divisions[0]) == type(df.index[0])
     tm.assert_frame_equal(df, ddf.compute())
+    ddf = dd.from_pandas(df, chunksize=8)
+    assert len(ddf.dask) == 3
+    assert len(ddf.divisions) == len(ddf.dask) + 1
+    assert type(ddf.divisions[0]) == type(df.index[0])
+    tm.assert_frame_equal(df, ddf.compute())
```
Comment (sinhrks) [inline dask/dataframe/tests/test_io.py]: Can you add test if it raises error expectedly, refer to other tests which uses ``assertRaisesRegexp``.

File: dask/dataframe/io.py (hunk 2 above)
Comment (thequackdaddy) [inline dask/dataframe/io.py]: I will change this. Note that the parenthesis format is elsewhere in the io fiie as well, namely in the `from_bcolz` function.

Comment (mrocklin) [inline dask/dataframe/io.py]: Yeah, I've been sloppy about following numpydoc explicitly in the past.  We
should endeavor to do this in the future though.  I'll add a TODO for
myself to go through an clean up existing docstrings.

File: dask/dataframe/tests/test_io.py (hunk 4 above)
Comment (thequackdaddy) [inline dask/dataframe/tests/test_io.py]: Ok. This is now a test. (For some reason, its not showing this diff is outdated... but it is... maybe I messed up something?)

**Comment by thequackdaddy** [comment]
I went ahead and updated the PR per @sinhrks  reuests. I also cheated and downloaded a git GUI to help me rebase... which I think I've done successfully. 

Have a look.

**Comment by mrocklin** [comment]
This looks OK to me.  I'll probably squash down commits.

@sinhrks any further comments?

File: dask/dataframe/io.py [hunk 5]
```diff
@@ -338,8 +338,11 @@ def from_pandas(data, npartitions, sort=True):
     ----------
     df : pandas.DataFrame or pandas.Series
         The DataFrame/Series with which to construct a dask DataFrame/Series
-    npartitions : int
+    npartitions : int, default None
```
Comment (thequackdaddy) [inline dask/dataframe/io.py]: No problem. 

The reason I went that route was that you have to pick one... so you can't option for neither. But I'll go with it. There's a check that you only have picked one anyways.

**Comment by thequackdaddy** [comment]
Squashed.

**Comment by jcrist** [comment]
Excellent, thanks for the PR:) Merging.

**Comment by thequackdaddy** [comment]
No. Thank you for such an excellent library.

**Comment by thequackdaddy** [comment]
Actually, this is my 2nd merged PR :smirk: 

That's fine. My name is Peter Quackenbush.

**Comment by sinhrks** [comment]
Thanks @thequackdaddy !

**Pull request merged**